tests/middleware/test_base.py

```diff
@@ -52,8 +52,9 @@ def test_custom_middleware(test_client_factory):
     response = client.get("/")
     assert response.headers["Custom-Header"] == "Example"
 
-    with pytest.raises(Exception):
+    with pytest.raises(Exception) as ctx:
```

Comment:
When running on `trio`, previously we would be getting an unexpected "Portal is not running" error here, and misinterpreting it as the `Exception()` in the `/exc` view. So I made the latter more precise and made sure we're explicitly checking for it.